client/app/pages/queries/view.js

```diff
@@ -42,7 +42,7 @@ function QueryViewCtrl(
     }
 
     $scope.showLog = false;
-    if ($scope.isDirty) {
+    if ($scope.isDirty || !isEmpty(selectedQueryText)) {
```

Comment:
Why is `isDirty` being checked here actually?

Reply:
If the query text wasn't changed we use the results API that takes just the query id (the safe API which requires only read access). If the text was changed, we use the results API that takes SQL (requires full access).